.github/workflows/test.yml

```diff
@@ -66,9 +71,48 @@ jobs:
           sudo curl -L https://github.com/docker/compose/releases/download/v2.26.0/docker-compose-`uname -s`-`uname -m` -o "/usr/local/lib/docker/cli-plugins/docker-compose"
           sudo chmod +x "/usr/local/lib/docker/cli-plugins/docker-compose"
 
+      - name: Prepare Docker Volume Caching
+        id: cache_key
+        run: |
+          # Set permissions for docker volumes so we can cache and restore
+          sudo chmod o+x /var/lib/docker
+          sudo chmod -R o+rwx /var/lib/docker/volumes
+          source .env
+          SENTRY_IMAGE_SHA=$(docker buildx imagetools inspect $SENTRY_IMAGE --format "{{println .Manifest.Digest}}")
+          echo "SENTRY_IMAGE_SHA=$SENTRY_IMAGE_SHA" >> $GITHUB_OUTPUT
+          SNUBA_IMAGE_SHA=$(docker buildx imagetools inspect $SNUBA_IMAGE --format "{{println .Manifest.Digest}}")
+          echo "SNUBA_IMAGE_SHA=$SNUBA_IMAGE_SHA" >> $GITHUB_OUTPUT
+
+      - name: Restore DB Volumes Cache
+        id: restore_cache
+        uses: actions/cache/restore@v4
+        with:
+          key: db-volumes-v4-${{ steps.cache_key.outputs.SENTRY_IMAGE_SHA }}-${{ steps.cache_key.outputs.SNUBA_IMAGE_SHA }}
+          restore-keys: |
+            db-volumes-v4-${{ steps.cache_key.outputs.SENTRY_IMAGE_SHA }}
+            db-volumes-v4-
```

Comment:
It does not seem like this is intentional? Perhaps 

`db-volumes-v4-${{ steps.cache_key.outputs.SNUBA_IMAGE_SHA }}` ?

Reply:
Completely intentional. These are prefix matched in order for stale cache restoration. So if we cannot match on both sentry and snuba image SHA, then we fallback to one with sentry image sha regardless of snuba, and finally, it just restores _any cache_ as long as it is v4.